langchain/chains/api/base.py

```diff
@@ -65,6 +67,19 @@ def validate_api_answer_prompt(cls, values: Dict) -> Dict:
             )
         return values
 
+    def verify_api_url_is_legit(self, parsed_url: str) -> None:
+        """Verify that the parsed URL corresponds to one of the URLs in the API spec"""
+
+        base_url_re = re.compile("http[s]?://((?:[-\w.]|(?:%[\da-fA-F]{2}))+)")
+        base_parsed_url = base_url_re.findall(parsed_url)[0].strip().lower()
+        api_doc_allowed_urls = base_url_re.findall(self.api_docs)
```

Comment:
we can probably find these ahead of time? or even make the user pass them in?

Reply:
Sure, i've pushed a fix.
The heavier regex (the once that's extracting allowed base URLs from API docs) is now running only once, on API Chain creation. 
@hwchase17 IMO this is the better option ("find these ahead of time", as you said), but if you prefer i can also expose an option for the user to pass the allowed base URLs (i just don't see a very good reason, because i see the API docs as the interface to do that)